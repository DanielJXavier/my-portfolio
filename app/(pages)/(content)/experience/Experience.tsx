"use client";

import { createContext, useState } from "react";

import Title from "@/components/Title";
import Work from "@/icons/Work";

import ExperienceItem from "./components/ExperienceItem";

import Menu from "../components/Menu";

import { experience } from "./data";

export const ExperienceModeContext = createContext("");

export default function Experience() {
  const [experienceMode, setExperienceMode] = useState("simple");

  return (
    <ExperienceModeContext.Provider value={experienceMode}>
      <Title icon={<Work />}>Experience</Title>
      <Menu
        items={["simple", "detailed"]}
        activeItem={experienceMode}
        setActiveItem={setExperienceMode}
      />
      {experience.map(
        (
          {
            role,
            companyId,
            companyName,
            year,
            description,
            responsibilities,
            biggestChallenge,
            hasBlackLogo,
          },
          i
        ) => (
          <ExperienceItem
            key={i}
            role={role}
            companyId={companyId}
            companyName={companyName}
            year={year}
            description={description}
            responsibilities={responsibilities}
            biggestChallenge={biggestChallenge}
            hasBlackLogo={hasBlackLogo}
          />
        )
      )}
    </ExperienceModeContext.Provider>
  );
}
