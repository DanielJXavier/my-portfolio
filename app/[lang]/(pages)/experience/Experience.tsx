"use client";

import { createContext, useRef, useState } from "react";

import { useParams } from "next/navigation";

import { ExperienceItemsType, getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import Title from "@/_components/Title";
import Work from "@/_icons/Work";

import Menu from "../_components/Menu";

import ExperienceItem from "./_components/ExperienceItem";
import { experience } from "./_config";

export const ExperienceModeContext = createContext("");

export default function Experience() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    experience: { title, menu, items, imageAltText },
  } = getDictionary(lang);

  const [experienceMode, setExperienceMode] = useState("simple");

  const menuItems = useRef([
    { text: menu.simple, value: "simple" },
    { text: menu.detailed, value: "detailed" },
  ]);

  const experienceItems = useRef<ExperienceItemsType>(items);

  return (
    <ExperienceModeContext.Provider value={experienceMode}>
      <Title icon={<Work />}>{title}</Title>

      <Menu
        items={menuItems.current}
        activeItem={experienceMode}
        handleClick={setExperienceMode}
      />

      {experience
        .filter(({ hidden }) => !hidden)
        .map(
          ({
            key,
            companyId,
            companyName,
            startDate,
            endDate,
            mainStack,
            hasBlackLogo,
          }) => (
            <ExperienceItem
              key={key}
              role={experienceItems.current[key].role}
              companyId={companyId}
              companyName={companyName}
              startDate={startDate}
              endDate={endDate}
              accomplishments={experienceItems.current[key].accomplishments}
              mainStack={mainStack}
              imageAltText={imageAltText}
              hasBlackLogo={hasBlackLogo}
            />
          )
        )}
    </ExperienceModeContext.Provider>
  );
}
