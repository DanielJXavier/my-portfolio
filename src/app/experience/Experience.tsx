"use client";

import { createContext, useState } from "react";

import Resposibilities from "./components/Responsibilities";
import GreatestChallenge from "./components/GreatestChallenge";
import ExperienceItem from "./components/ExperienceItem";

export const ExperienceContext = createContext("");

export default function Experience() {
  const [experience, setExperience] = useState("simple");

  return (
    <ExperienceContext.Provider value={experience}>
      <main className="container mx-auto px-4 md:px-5 xl:px-6 pt-6 md:pt-8 xl:pt-12">
        <h1 className="text-xl md:text-2xl lg:text-3xl mb-4 md:mb-6 xl:mb-8 flex items-center gap-x-3">
          <span>Experience</span>
          <span className="text-sm lg:text-base">
            (
            <button
              className={`${experience === "simple" ? "underline" : ""}`}
              onClick={() => setExperience("simple")}
            >
              simple
            </button>{" "}
            |{" "}
            <button
              className={`${experience === "detailed" ? "underline" : ""}`}
              onClick={() => setExperience("detailed")}
            >
              detailed
            </button>
            )
          </span>
        </h1>
        <ExperienceItem
          logoPath="/grupoolx_logo.jpeg"
          year="2024"
          role="Software Engineering Manager"
          company="Grupo OLX"
          description="I was a leader of the &ldquo;Ads Seller Experience&rdquo; team."
        >
          <Resposibilities>
            <li>
              To directly lead all developers (9 devs) on the team, acting as a
              reference;
            </li>
            <li>
              To ensure delivery of the platform&apos;s registration and ad
              management system roadmap;
            </li>
            <li>To bring engagement to the team;</li>
            <li>
              To do &ldquo;1:1s&rdquo; and help those led to execute their
              performance cycle;
            </li>
            <li>To apply agile methodologies and conduct team ceremonies;</li>
            <li>
              To support the Product Manager to make the best decisions for the
              product;
            </li>
          </Resposibilities>
          <GreatestChallenge>
            To handle the high expectations of leaders for short-term
            deliveries.
          </GreatestChallenge>
        </ExperienceItem>
        <ExperienceItem
          logoPath="/grupoolx_logo.jpeg"
          year="2023"
          role="Software Engineering Manager"
          company="Grupo OLX"
          description="I was a leader of the &ldquo;Search & Recommendation&rdquo; team&sup1; and the &ldquo;Design System&rdquo; chapter&sup2;."
        >
          <Resposibilities>
            <li>
              To directly lead all developers (8 devs) on the team, acting as a
              reference;&sup1;
            </li>
            <li>
              To ensure delivery of the search and recommendation systems
              roadmap;&sup1;
            </li>
            <li>
              To organize the team and propose a better structure to support the
              strategic plan for the next year;&sup1;
            </li>
            <li>
              To clarify the company&apos;s understanding of the search and
              recommendation subjects;&sup1;
            </li>
            <li>
              To indirectly lead all developers (9 devs) on the chapter;&sup2;
            </li>
            <li>
              To bring engagement to the chapter and make valuable deliveries by
              working just once a week;&sup2;
            </li>
          </Resposibilities>
          <GreatestChallenge>
            To handle both big responsibilities (team&sup1; and chapter&sup2;)
            at the same time.
          </GreatestChallenge>
        </ExperienceItem>
        <ExperienceItem
          logoPath="/grupoolx_logo.jpeg"
          year="2022~2023"
          role="Software Engineering Manager"
          company="Grupo OLX"
          description="I was a leader of the &ldquo;Design System&rdquo; team."
        >
          <Resposibilities>
            <li>
              To directly lead all developers (9 devs) on the team, acting as a
              reference;
            </li>
            <li>
              To partner with the Design Manager to make the best decisions for
              the product and create a roadmap;
            </li>
            <li>To ensure delivery of the Design System script;</li>
            <li>
              To collect and analyze data to ensure the project was valuable to
              the company;
            </li>
            <li>
              To ensure that the Design System is evolving and is used by other
              teams;
            </li>
            <li>
              To create documentations and processes for contributing to the
              Design System project and managing its version;
            </li>
          </Resposibilities>
          <GreatestChallenge>
            To perform several roles (Engineering Manager, Product Manager, Data
            Analyst, etc.) at the same time.
          </GreatestChallenge>
        </ExperienceItem>
        <ExperienceItem
          logoPath="/grupoolx_logo.jpeg"
          year="2021~2022"
          role="Software Engineering Manager"
          company="Grupo OLX"
          description="I was a leader of the &ldquo;Advertising&rdquo; team."
        >
          <Resposibilities>
            <li>
              To directly lead all developers (7 devs) on the team, acting as a
              reference;
            </li>
            <li>
              To ensure delivery of the platform&apos;s direct and indirect
              Advertising roadmap;
            </li>
            <li>
              To hire new professionals to get the necessary productive
              capacity;
            </li>
            <li>
              To partner actively with the Product Manager to prioritize and
              develop the best initiatives;
            </li>
            <li>
              To propose a new structure for the Advertising area to get better
              results;
            </li>
          </Resposibilities>
          <GreatestChallenge>
            To manage high expectations regarding Advertising revenue results.
          </GreatestChallenge>
        </ExperienceItem>
      </main>
    </ExperienceContext.Provider>
  );
}
