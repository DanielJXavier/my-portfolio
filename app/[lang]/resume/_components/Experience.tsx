import { useParams } from "next/navigation";

import { ExperienceItemsType, getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import { experience } from "@/[lang]/(pages)/experience/_config";
import { hardSkills } from "@/[lang]/(pages)/skills/_config";

import VisibilityControl from "./VisibiltyControl";
import { useRef } from "react";

export default function Experience() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    experience: {
      items,
      previousItemsText,
      biggestChallengeTitle,
      mainStackTitle,
    },
    resume: { sectionTitles },
  } = getDictionary(lang);

  const experienceItems = useRef<ExperienceItemsType>(items);

  return (
    <section>
      <h2 className="mb-2 text-[16pt] font-bold">{sectionTitles.experience}</h2>
      <ul className="flex flex-col gap-y-3">
        {experience.map(
          ({
            key,
            companyName,
            year,
            mainStack,
            resumeAccomplishments,
            resume,
          }) => (
            <VisibilityControl
              key={key}
              className="items-baseline break-inside-avoid"
              defaultVisible={resume}
            >
              <div>
                <p className="text-[12pt]">
                  <span className="font-bold">
                    {experienceItems.current[key].role} @ {companyName}
                  </span>
                  <span> ({year})</span>
                </p>
                <ul className="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['â€¢'] [&_li]:before:absolute [&_li]:before:-left-[10px]">
                  {experienceItems.current[key].accomplishments.map(
                    (accomplishment, i) => (
                      <VisibilityControl
                        key={`accomplishment-${i}`}
                        className="items-baseline"
                        defaultVisible={resumeAccomplishments?.some(
                          (resumeAccomplishment) =>
                            resumeAccomplishment === i + 1
                        )}
                      >
                        {accomplishment}
                      </VisibilityControl>
                    )
                  )}
                  <VisibilityControl
                    className="mt-1 items-baseline"
                    defaultVisible={resume}
                  >
                    {biggestChallengeTitle}:{" "}
                    {experienceItems.current[key].biggestChallenge}
                  </VisibilityControl>
                  <VisibilityControl
                    className="mt-1 items-baseline"
                    defaultVisible={resume}
                  >
                    {mainStackTitle}:{" "}
                    {hardSkills
                      .filter(({ key }) =>
                        mainStack.some((stack) => stack === key)
                      )
                      .map(({ name }) => name)
                      .join(" | ")}
                  </VisibilityControl>
                </ul>
              </div>
            </VisibilityControl>
          )
        )}
        <li>
          {previousItemsText.prefix} {previousItemsText.sufix}
        </li>
      </ul>
    </section>
  );
}
