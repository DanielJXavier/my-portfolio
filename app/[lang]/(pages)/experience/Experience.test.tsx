import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import { useParams } from "next/navigation";

import Experience from "./Experience";

jest.mock("next/navigation");

(useParams as jest.Mock).mockImplementation(() => ({ lang: "en" }));

describe("Experience page (Client-side)", () => {
  it("renders the page", () => {
    const { container } = render(<Experience />);

    expect(container).toMatchSnapshot();
  });
});
