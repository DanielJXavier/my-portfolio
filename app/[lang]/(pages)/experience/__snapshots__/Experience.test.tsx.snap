// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience page (Client-side) renders the page 1`] = `
<div>
  <h1
    class="text-xl md:text-2xl lg:text-3xl flex items-center gap-x-2"
    data-testid="title"
  >
    <svg
      class="fill-primary size-6 md:size-8"
      viewBox="0 -960 960 960"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M157.69-140q-23.53 0-40.61-17.08T100-197.69v-444.62q0-23.53 17.08-40.61T157.69-700H340v-82.3q0-23.53 17.08-40.62Q374.16-840 397.69-840h164.62q23.53 0 40.61 17.08Q620-805.83 620-782.3v82.3h182.31q23.53 0 40.61 17.08T860-642.31v444.62q0 23.53-17.08 40.61T802.31-140H157.69Zm0-45.39h644.62q4.61 0 8.46-3.84 3.84-3.85 3.84-8.46v-444.62q0-4.61-3.84-8.46-3.85-3.84-8.46-3.84H157.69q-4.61 0-8.46 3.84-3.84 3.85-3.84 8.46v444.62q0 4.61 3.84 8.46 3.85 3.84 8.46 3.84ZM385.39-700h189.22v-82.3q0-4.62-3.84-8.47-3.85-3.84-8.46-3.84H397.69q-4.61 0-8.46 3.84-3.84 3.85-3.84 8.47v82.3Zm-240 514.61V-654.61-185.39Z"
      />
    </svg>
    Experience
  </h1>
  <menu
    class="mt-1 md:mt-2 text-sm lg:mt-3 lg:text-base flex gap-x-1"
    data-testid="menu"
  >
    <li>
      <button
        class="text-secondary underline cursor-default hover:text-secondary hover:underline"
        disabled=""
      >
        simple
      </button>
    </li>
    <li>
      |
    </li>
    <li>
      <button
        class=" hover:text-secondary hover:underline"
      >
        detailed
      </button>
    </li>
  </menu>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-secondary before:via-secondary before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of <DanielXavier /> company"
        class="rounded-full border-2 border-solid border-secondary"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fdanielxavier.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fdanielxavier.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fdanielxavier.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Senior Front-End Engineer | Tech Lead
         @ 
        &lt;DanielXavier /&gt;
        <span
          class="text-xs md:text-sm"
        >
           
          (
          05-2024
           
          to
           
          present
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        A contractor role where I have been working on some independent client projects.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-secondary before:via-secondary before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of PDSAcademy company"
        class="rounded-full border-2 border-solid border-secondary"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fpdsacademy.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fpdsacademy.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fpdsacademy.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Post-graduation professor
         @ 
        PDSAcademy
        <span
          class="text-xs md:text-sm"
        >
           
          (
          06-2025
           
          to
           
          06-2025
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        A short-term contract role where I teach Front-End development.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-[#6d0ad5] before:via-[#6d0ad5] before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of OLX company"
        class="rounded-full "
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Folx.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Folx.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Folx.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Technical Lead
         @ 
        OLX
        <span
          class="text-xs md:text-sm"
        >
           
          (
          08-2021
           
          to
           
          04-2024
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        A tech leader role where I was responsible for different core teams.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-[#8568ec] before:via-[#8568ec] before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of Meaning company"
        class="rounded-full "
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fmeaning.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fmeaning.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fmeaning.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        CTO & Partner
         @ 
        Meaning
        <span
          class="text-xs md:text-sm"
        >
           
          (
          03-2022
           
          to
           
          05-2023
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        I held the position of CTO and co-founder at the startup.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-[#5d5ef1] before:via-[#5d5ef1] before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of Softplan company"
        class="rounded-full "
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fsoftplan.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fsoftplan.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fsoftplan.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Team Leader
         @ 
        Softplan
        <span
          class="text-xs md:text-sm"
        >
           
          (
          10-2020
           
          to
           
          07-2021
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        A multidisciplinary team leader role where I was responsible for the team results.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-[#5d5ef1] before:via-[#5d5ef1] before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of Softplan company"
        class="rounded-full "
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fsoftplan.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fsoftplan.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fsoftplan.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Technical Lead
         @ 
        Softplan
        <span
          class="text-xs md:text-sm"
        >
           
          (
          09-2019
           
          to
           
          10-2020
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        A tech leader position (80% tech / 20% people) where I was responsible for the Front-End best practices.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-[#5d5ef1] before:via-[#5d5ef1] before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of Softplan company"
        class="rounded-full "
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fsoftplan.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fsoftplan.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fsoftplan.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Senior Front-End Engineer
         @ 
        Softplan
        <span
          class="text-xs md:text-sm"
        >
           
          (
          12-2018
           
          to
           
          08-2019
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        A pure hands-on position where I was responsible for doing all of my team's Front-End deliveries.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-[#ee4e21] before:via-[#ee4e21] before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of Hotmart company"
        class="rounded-full "
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fhotmart.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fhotmart.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fhotmart.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Senior Full-Stack Engineer
         @ 
        Hotmart
        <span
          class="text-xs md:text-sm"
        >
           
          (
          06-2018
           
          to
           
          11-2018
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        A pure hands-on position where I was responsible for doing all of my team's full-stack deliveries.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-[#ee4e21] before:via-[#ee4e21] before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of Hotmart company"
        class="rounded-full "
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fhotmart.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fhotmart.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fhotmart.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Mid Level Front-End Engineer
         @ 
        Hotmart
        <span
          class="text-xs md:text-sm"
        >
           
          (
          06-2017
           
          to
           
          05-2018
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        A pure hands-on position where I was responsible for doing some of my team's full-stack deliveries.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-[#ee4e21] before:via-[#ee4e21] before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of Hotmart company"
        class="rounded-full "
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fhotmart.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fhotmart.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fhotmart.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Entry Level Front-End Engineer
         @ 
        Hotmart
        <span
          class="text-xs md:text-sm"
        >
           
          (
          12-2016
           
          to
           
          05-2017
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        I was an Entry Level Front-End developer on the "User Acquisition" team.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-secondary before:via-secondary before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of Sydle company"
        class="rounded-full border-2 border-solid border-secondary"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fsydle.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fsydle.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fsydle.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Support Analyst
         @ 
        Sydle
        <span
          class="text-xs md:text-sm"
        >
           
          (
          04-2014
           
          to
           
          11-2016
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        I was a Support Analyst on the "Suporte & Sustentação" team.
      </h3>
    </section>
  </article>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-secondary before:via-secondary before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of Sydle company"
        class="rounded-full border-2 border-solid border-secondary"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fsydle.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fsydle.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fsydle.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Support Associate
         @ 
        Sydle
        <span
          class="text-xs md:text-sm"
        >
           
          (
          04-2012
           
          to
           
          03-2014
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        I was a Support Associate on the "Suporte & Sustentação" team.
      </h3>
    </section>
  </article>
</div>
`;
